crates/democracy/src/lib.rs

```diff
@@ -964,37 +961,18 @@ impl<T: Config> Pallet<T> {
 
     /// determine whether or not a new referendum should be launched. This will return true
     /// once every week.
-    fn should_launch(now: Duration) -> Result<bool, DispatchError> {
-        if now.as_secs() < NextLaunchTimestamp::<T>::get() {
-            return Ok(false);
+    fn should_launch(now: u64) -> bool {
+        if now < NextLaunchTimestamp::<T>::get() {
+            return false;
         }
 
-        // time to launch - calculate the date of next launch.
-
-        // convert to format used by `chrono`
-        let secs: i64 = now.as_secs().try_into().map_err(Error::<T>::from)?;
-        let now =
-            chrono::NaiveDateTime::from_timestamp_opt(secs, now.subsec_nanos()).ok_or(Error::<T>::TryIntoIntError)?;
-
-        // calculate next week boundary
-        let beginning_of_week = now.date().week(chrono::Weekday::Mon).first_day();
-        let next_week = beginning_of_week
-            .checked_add_days(Days::new(7))
-            .ok_or(Error::<T>::TryIntoIntError)?
-            .and_time(Default::default());
-
-        let offset = T::LaunchOffsetMillis::get()
-            .try_into()
-            .map_err(|_| Error::<T>::TryIntoIntError)?;
-        let next_launch = next_week
-            .checked_add_signed(chrono::Duration::milliseconds(offset))
-            .ok_or(ArithmeticError::Overflow)?;
-
         // update storage
-        let next_timestamp: u64 = next_launch.timestamp().try_into().map_err(Error::<T>::from)?;
-        NextLaunchTimestamp::<T>::set(next_timestamp);
+        NextLaunchTimestamp::<T>::mutate(|next_launch_timestamp| {
+            // offset is number of seconds - e.g. to next week (mon 9am)
+            next_launch_timestamp.saturating_accrue(T::LaunchOffset::get());
```

Comment:
If our chain gets stalled for a while, this can cause multiple rounds of proposals to be opened. If we want to prevent that, we can do something like `next_launch_timestamp = next_launch_timestamp + ((now - next_launch_timestamp) / LaunchOffset + 1) * LaunchOffset`. Or we can just accept the behavior, but we need to make a decision on that